src/insert/spacefinder/spacefinder.ts

```diff
@@ -198,22 +203,79 @@ const partitionCandidates = <T>(
+ *                                                        │
+ *                  ───────────────────  Top of container │          ───────────────────  Top of container
+ *                    ▲              ▲                    │            ▲              ▲
+ *                    │              │                    │            │              │ candidate.top
+ *                    │ ┌──────────┐ │                    │            │ ┌──────────┐ ▼   (insertion point)
+ *                    │ │          │ │opponent.bottom     │            │ │          │
+ *                    │ │ Opponent │ │                    │            │ │ Candidate│
+ *       candidate.top│ │          │ │                    │opponent.top│ │          │
+ *                    │ └──────────┘ ▼                    │            │ └──────────┘
+ *                    │           ▲                       │            │
+ *                    │           │ minBelow              │            │  ───────────
+ *                    │           ▼                       │            │           ▲
+ *                    │ ────────────                      │            │           │ minAbove
+ *                    │                                   │            │           ▼
+ * (insertion point)  ▼ ┌──────────┐                      │            ▼ ┌──────────┐
+ *                      │          │                      │              │          │
+ *                      │ Candidate│                      │              │ Opponent │
+ *                      │          │                      │              │          │
+ *                      └──────────┘                      │              └──────────┘
+ *                                                        │
+ *                                                        │
+ *
+ * @param candidate
+ * @param opponent
+ * @param rule
+ * @param isCandidateBelow
+ * @returns
+ */
+const isTopOfCandidateFarEnoughFromOpponent = (
+	candidate: SpacefinderItem,
+	opponent: SpacefinderItem,
+	rule: RuleSpacing,
+	isCandidateBelow: boolean,
+): boolean => {
+	const potentialInsertPosition = candidate.top;
+
+	if (isCandidateBelow && rule.minBelow) {
+		return opponent.bottom + rule.minBelow <= potentialInsertPosition;
+	}
+
+	if (!isCandidateBelow && rule.minAbove) {
+		return opponent.top - rule.minAbove >= potentialInsertPosition;
+	}
+
+	// if no rule is set (or they're 0), return true
+	return true;
+};
+
 // test one element vs another for the given rules
 const testCandidate = (
 	rule: RuleSpacing,
 	candidate: SpacefinderItem,
 	opponent: SpacefinderItem,
 ): boolean => {
-	const isMinAbove = candidate.top - opponent.bottom >= rule.minAbove;
-	const isMinBelow = opponent.top - candidate.top >= rule.minBelow;
-
-	const pass = isMinAbove || isMinBelow;
+	const isCandidateBelow =
+		candidate.top > opponent.top && candidate.bottom > opponent.bottom;
+
+	const pass = isTopOfCandidateFarEnoughFromOpponent(
```

Comment:
We can probably can get away with slightly shortening this to `isCandidateFarEnoughFromOpponent`. The fact that the top of the candidate is used can be an implementation detail

Reply:
I think it's a good reminder that ads are inserted at the top of paragraphs (I went into this think they were inserted at the bottom!)